packages/stateless/components/modals/NftSelectionModal.tsx

```diff
@@ -29,6 +39,9 @@ export interface NftSelectionModalProps<T extends NftCardInfo>
   actionLabel: string
   allowSelectingNone?: boolean
   selectedDisplay?: ReactNode
+  mode?: StakingMode
+  setMode?: (mode: StakingMode) => void
+  hasStake?: boolean
 }
```

Comment:
with the goal of keeping this component sufficiently generic, we should probably replace these with a generic `headerDisplay?: ReactNode` prop and define this logic where it's used